File: packages/audio/ios/Model/ArtistModel.swift
import MediaPlayer

struct ArtistModel: ArtistModelContract {
    let media: MPMediaItemCollection

    var id: ArtistId {
        ArtistId(id: media.representativeItem?.artistPersistentID ?? 0)
    }

    var name: String {
        media.representativeItem?.artist ?? "undefined"
    }
}

File: model/ArtistModel.ts
import { ArtistPayload } from "audio";

export class ArtistModel {
  id: string;
  name: string;

  constructor(
    id: string,
    name: string,
  ) {
    this.id = id;
    this.name = name;
  }

  static fromPayload(artist: ArtistPayload): ArtistModel {
    return new ArtistModel(
      artist.id,
      artist.name,
    );
  }
}
